helpdesk_client/zendesk_manager.py

```
@@ -0,0 +1,280 @@
+
+from helpdesk_client.interfaces import (
+    HelpDeskBase,
+    HelpDeskComment,
+    HelpDeskCustomField,
+    HelpDeskException,
+    HelpDeskTicket,
+    HelpDeskTicketNotFoundException,
+    HelpDeskUser,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class ZendeskClientNotFoundException(Exception):
+    pass
+
+
+class ZendeskManager(HelpDeskBase):
+    def __init__(self, **kwargs):
+        """Create a new Zendesk client - pass credentials in
+
+        :param credentials: The credentials required to create client { token , email, subdomain }.
+        """
+        if not kwargs.get("credentials", None):
+            raise ZendeskClientNotFoundException("No Zendesk credentials provided")
+
+        self.client = Zenpy(
+            timeout=kwargs.get("credentials").get("timeout", 5),
+            email=kwargs.get("credentials")["email"],
+            token=kwargs.get("credentials")["token"],
+            subdomain=kwargs.get("credentials")["subdomain"],
+        )
+
+    def get_or_create_user(self, user: HelpDeskUser = None) -> HelpDeskUser:
+        """Get or Create a new Zendesk user   /PS-IGNORE
+
+        :param HelpDeskUser
+                full_name: string full name for Zendesk user.
+                email: string email address text for the Zendesk user.
+
+        :returns: HelpDeskUser instance representing Zendesk user.
+        """
+
+        transformed_user = self.__transform_helpdesk_user_to_zendesk_user(user)
+
+        if transformed_user is None:
+            logger.error("No Zendesk user to retrieve")  # Error log /PS-IGNORE,
+            return None
+        elif transformed_user.id:
+            zendesk_user = self.client.users(id=transformed_user.id)
+        else:
+            zendesk_user = self.client.users.create_or_update(transformed_user)
+
+        if zendesk_user is None:
+            logger.error("No Zendesk user found")  # Error log /PS-IGNORE,
+            return None
```

Same as the previous comment, `__transform_helpdesk_user_to_zendesk_user` should raise an exception.

---

updating to raise exception